app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
   belongs_to :hotel
+  has_many :guest_rooms
+  has_many :guests, through: :guest_rooms
+
+  def guest_count
+    guests.count
+  end
+
+  def guests_distinct
```

Comment:
This works just fine, but another option that is also VERY magical: putting this method in the Hotel model
You might be thinking.. how would the hotel model have an attribute called `guests`? And it wouldn't but you could add an additional association to hotel to allow it mooch off the associations set up for rooms. Rooms belong to hotels, and rooms has the association to guests, so if you add `has_many :guests, through: :rooms` to the hotel model, you can actually do `hotel.guests`! It's magic! Try it out